src/table/stories/app-subclassing-filter-table.component.ts

```diff
@@ -0,0 +1,183 @@
+				(checkedChange)="filterCountries('US', $event)">
+				US
+			</ibm-checkbox>
+			<ibm-checkbox
+				[checked]="displayedCountries.includes('France')"
+				(checkedChange)="filterCountries('France', $event)">
+				France
+			</ibm-checkbox>
+			<ibm-checkbox
+				[checked]="displayedCountries.includes('Argentina')"
+				(checkedChange)="filterCountries('Argentina', $event)">
+				Argentina
+			</ibm-checkbox>
+			<ibm-checkbox
+				[checked]="displayedCountries.includes('Japan')"
+				(checkedChange)="filterCountries('Japan', $event)">
+				Japan
+			</ibm-checkbox>
+		</div>
+	</ng-template>
+	`
+})
+export class FilterWithSubclassingStory implements OnInit {
+	@Input() size = "md";
+	@Input() showSelectionColumn = true;
+	@Input() enableSingleSelect = false;
+	@Input() striped = true;
+	@Input() isDataGrid = false;
+	@Input() noData = false;
+	@Input() stickyHeader = false;
+	@Input() skeleton = false;
+
+	model = new TableModel();
+	displayedCountries = ["US", "France", "Argentina", "Japan"];
+	searchValue = "";
+
+	dataset = [
+		[
+			new TableItem({ data: "800" }),
+			new TableItem({ data: "East Sadye" }),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "US" })
+		],
+		[
+			new TableItem({ data: "500" }),
+			new TableItem({ data: "Lueilwitzview" }),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "US" })
+		],
+		[
+			new TableItem({ data: "120" }),
+			new TableItem({ data: "East Arcelyside" }),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "France" })
+		],
+		[
+			new TableItem({ data: "119" }),
+			new TableItem({data: "West Dylan"}),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "Argentina" })
+		],
+		[
+			new TableItem({ data: "54" }),
+			new TableItem({data: "Brandynberg"}),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "Japan" })
+		],
+		[
+			new TableItem({ data: "15" }),
+			new TableItem({data: "Stoltenbergport"}),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "Canada" })
+		],
+		[
+			new TableItem({ data: "12" }),
+			new TableItem({data: "Rheabury"}),
+			new TableItem({ data: "Store" }),
+			new TableItem({ data: "US" })
+		]
+	];
+
+	filterNodeNames(searchString: string) {
+		this.searchValue = searchString;
+	}
+
+	filterCountries(countryName: string, checked: boolean) {
+		if (checked) {
+			this.displayedCountries.push(countryName);
+		} else {
+			this.displayedCountries.splice(this.displayedCountries.indexOf(countryName), 1);
+		}
+	}
+
+	overflowOnClick = (event: any) => {
+		event.stopPropagation();
+	}
+
+	ngOnInit() {
+		this.model.header = [
+			new TableHeaderItem({
+				data: "Node ID"
+			}),
+			new TableHeaderItem({
+				data: "Node name"
+			}),
+			new TableHeaderItem({
+				data: "Node type"
+			}),
+			new TableHeaderItem({
+				data: "Country"
+			})
+		];
+
+		this.model.data = this.dataset;
+
+		this.model.isRowFiltered = (index: number) => {
```

Comment:
This is not really subclassing but it's probably better and the best way overall because it doesn't require using an outside data storage. Can we mark it recommended in the title (in addition to changing the subclassing part)?